src/app/meditation/meditation.component.ts

```diff
@@ -384,6 +433,17 @@ export class MeditationComponent {
         },
         err => console.error(err)
       );
+
+    // get commitment status for user
+    this.commitmentService.getCurrentUser()
+      .map(res => res.json())
+      .subscribe(
```

Comment:
You need to merge the Observables `userService.getProfile` and `commitmentService.getCurrentUser` via `Observable.merge`. Otherwise `this.commitmentProgress = this.reached(data);` will return `undefined` when the profile isn't loaded.